cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -136,6 +137,40 @@ class ARROW_DS_EXPORT ParquetFileFormat : public FileFormat {
       fs::FileLocator destination_locator) const override;
 
   std::shared_ptr<FileWriteOptions> DefaultWriteOptions() override;
+
+  /// \brief A getter function to retrieve the dataset encryption configuration
+  std::shared_ptr<DatasetEncryptionConfiguration> GetDatasetEncryptionConfig() const {
+#ifdef PARQUET_REQUIRE_ENCRYPTION
+    return dataset_encryption_config_;
+#else
+    return NULLPTR;
+#endif
+  }
+  /// \brief A getter function to retrieve the dataset decryption configuration
+  std::shared_ptr<DatasetDecryptionConfiguration> GetDatasetDecryptionConfig() const {
+#ifdef PARQUET_REQUIRE_ENCRYPTION
+    return dataset_decryption_config_;
+#else
+    return NULLPTR;
+#endif
```

Comment:
It seems that these two functions are always called in the scope of `PARQUET_REQUIRE_ENCRYPTION`. So we can remove the branch inside, right?